src/js/zqd/zqd.js

```diff
@@ -124,6 +124,15 @@ export class ZQD {
       "-zeekrunner",
       zeekRunnerCommand(this.zeekRunner)
     ]
+
+    // For unix systems, pass posix pipe read file descriptor into zqd process.
+    // In the event of Brim getting shutdown via `SIGKILL`, this will let zqd
+    // know that it has been orphaned and to shutdown.
+    if (process.platform !== "win32") {
+      const {readfd} = require("node-pipe").pipeSync()
+      opts.stdio.push(readfd)
+      args.push(`-brimfd=${opts.stdio.length - 1}`)
```

Comment:
I'm not sure how this works so I may be off, but it looks like you'll pass in the integer 3 here. Were you intending to access the array `opts.stdio[opts.stdio.length - 1]`?

Reply:
This is intended. See https://github.com/brimsec/brim/pull/1031#discussion_r478654813